integration/tests/basic/test_rpc_calls.py

```diff
@@ -208,6 +208,34 @@ def test_eth_get_logs_blockhash(self, event_caller_contract, param_fields):
                                "blockNumber", "transactionIndex", "address",
                                "logIndex", "data", "transactionLogIndex"])
         assert_fields_are_boolean(response["result"][0], ["removed"])
+        assert_equal_fields(response, receipt, ["transactionHash", "blockHash",
+                                                "blockNumber", "transactionIndex", "address",
+                                                "logIndex", "data", "transactionLogIndex"])
+
+    def test_eth_get_logs_blockhash_empty_params(self, event_caller_contract):
+        number = random.randint(1, 100)
+        text = "".join([random.choice(string.ascii_uppercase) for _ in range(5)])
+        bytes_array = text.encode().ljust(32, b'\0')
+        bol = True
+        tx = self.make_tx_object()
+        instruction_tx = event_caller_contract.functions.allTypes(
+            self.sender_account.address, number, text, bytes_array, bol
+        ).build_transaction(tx)
+        receipt = self.web3_client.send_transaction(self.sender_account, instruction_tx)
+
+        params = {"blockHash": receipt["blockHash"].hex()}
+
+        response = self.proxy_api.send_rpc("eth_getLogs", params=params)
+        assert "error" not in response
+
+        assert_fields_are_hex(response["result"][0],
+                              ["transactionHash", "blockHash",
+                               "blockNumber", "transactionIndex", "address",
+                               "logIndex", "transactionLogIndex"])
+        assert_fields_are_boolean(response["result"][0], ["removed"])
+        assert_equal_fields(response, receipt, ["transactionHash", "blockHash",
```

Comment:
assert_equal_fields can't be checked only for blockHash field for that case

Reply:
Fixed